app/models/miq_server/log_management.rb

```diff
@@ -111,12 +111,12 @@ def synchronize_logs(*args)
   end
 
   def last_log_sync_on
-    rec = LogFile.where(:resource_type => self.class.name, :resource_id => id).order("updated_on DESC").select("updated_on").first
+    rec = LogFile.where(:resource => self).order("updated_on DESC").select("updated_on").first
```

Comment:
This should just be using the `log_files` association. (I haven't checked the others, but any of the form `ModelClass.where(:something => obj)`, where obj is known to not be nil (e.g., because we were confidently calling `.id` on it) should instead be `relevant_association`.

`new` too.